ERCS/erc-7887.md

```diff
@@ -0,0 +1,384 @@
+      type: address
+  outputs:
+```
+
+#### `cancelRedeemRequest`
+
+Submits a Request for asynchronous redeem cancelation. This places the Request in Pending state, with a corresponding increase in `pendingCancelRedeemRequest` for the full amount of the pending redeem Request. 
+
+When the cancelation is Pending, new redeem Requests are blocked and `requestRedeem` MUST revert.
+
+When the cancelation is Claimable, `claimableCancelRedeemRequest` will be increased for the `controller`. `claimCancelRedeemRequest` can subsequently be called by `controller` to receive `shares`. A Request MAY transition straight to Claimable state but MUST NOT skip the Claimable state.
+
+`controller` MUST equal `msg.sender` unless the `controller` has approved the `msg.sender` as an operator.
+
+MUST emit the `CancelRedeemRequest` event.
+
+```yaml
+- name: cancelRedeemRequest
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: requestId
+      type: uint256
+    - name: controller
+      type: address
+  outputs:
+```
+
+#### `pendingCancelRedeemRequest`
+
+Whether the given `requestId` and `controller` have a pending redeem cancelation Request.
+
+MUST NOT show any variations depending on the caller.
+
+MUST NOT revert unless due to integer overflow caused by an unreasonably large input.
+
+```yaml
+- name: pendingCancelRedeemRequest
+  type: function
+  stateMutability: view
+
+  inputs:
+    - name: requestId
+      type: uint256
+    - name: controller
+      type: address
+
+  outputs:
+    - name: isPending
+      type: bool
+```
+
+#### `claimableCancelRedeemRequest`
+
+The amount of `shares` in Claimable cancelation state for the `controller` to claim.
+
+MUST NOT show any variations depending on the caller.
+
+MUST NOT revert unless due to integer overflow caused by an unreasonably large input.
+
+```yaml
+- name: claimableCancelRedeemRequest
+  type: function
+  stateMutability: view
+
+  inputs:
+    - name: requestId
+      type: uint256
+    - name: controller
+      type: address
+
+  outputs:
+    - name: shares
+      type: uint256
+```
+
+#### `claimCancelRedeemRequest`
+
+Claims the redeem cancelation Request with `requestId` and `controller`.
+
+Transfers `assets` to `receiver`.
+
+`controller` MUST equal `msg.sender` unless the `controller` has approved the `msg.sender` as an operator.
+
+MUST emit the `ClaimCancelRedeemRequest` event.
+
+```yaml
+- name: claimCancelRedeemRequest
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: requestId
+      type: uint256
+    - name: receiver
+      type: address
+    - name: owner
+      type: address
+  outputs:
+```
+
+### Events
+
+#### `CancelDepositRequest`
+
+`controller` has requested cancelation of their deposit Request with request ID `requestId`. `sender` is the caller of the `cancelDepositRequest` which may not be equal to the `controller`.
+
+MUST be emitted when a deposit cancelation Request is submitted using the `cancelDepositRequest` method.
```

Comment:
Not a blocker, but it might be better to word this as:

```suggestion
MUST be emitted when a deposit cancelation Request is submitted, for example when using the `cancelDepositRequest` method.
```

Just in case certain implementations provide alternate ways to request cancelation.